pkgs/top-level/all-packages.nix

```diff
@@ -17579,6 +17579,10 @@ with pkgs;
     jre = jre8;
   };
 
+  inherit (callPackage ../applications/editors/jupyter-kernels/xeus-cling { })
+    cpp11-kernel cpp14-kernel cpp17-kernel cpp2a-kernel;
+  xeus-cling = callPackage ../applications/editors/jupyter-kernels/xeus-cling/xeus-cling.nix { };
```

Comment:
same remark as https://github.com/NixOS/nixpkgs/pull/171202/files#r1397979118

Reply:
I am annoyed to see those kernels at the top-level still since we dont make much of it without jupyter. Would it make sense to have the kernels in a package set like luaPackages/pythonPackages -> jupyterKernels = { xeus-cling , coq_kernel, ihaskell ... }. Is that on your roadmap. What do you think ?

Reply:
Yeah, I did argue just now against exposing kernels at the top-level :)

It's definitely on my roadmap. I could either do it in this PR or a subsequent one. Maybe a subsequent one would be cleaner? But I'm up for whatever you prefer.